File: 2181.cpp
struct Solution {
    ListNode* mergeNodes(ListNode* head) {
        ListNode foo(0, head);
        ListNode *s = &foo, *ptr = head;
        int accum = 0;
        while (ptr != nullptr){
            if (ptr->val != 0){
                accum += ptr->val;
            } else {
                ptr->val = accum;
                accum = 0;
                s->next = ptr;
                s = ptr;
            }
            ptr = ptr->next;
        }
        return foo.next->next;
    }
};

File: sol/2181/2181.py
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        foo = ListNode(0, head)
        s = foo
        ptr = foo.next
        accum = 0
        while ptr is not None:
            if ptr.val != 0:
                accum += ptr.val
            else:
                ptr.val = accum
                s.next = ptr
                s = ptr
                accum = 0
            ptr = ptr.next
        return foo.next.next
